composer/2022_airflow_summit/data_analytics_process.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import sys
+
+
+from py4j.protocol import Py4JJavaError
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col
+
+
+if __name__ == "__main__":
+    BUCKET_NAME = sys.argv[1]
+    READ_TABLE = sys.argv[2]
+    WRITE_TABLE = sys.argv[3]
+
+    # Create a SparkSession, viewable via the Spark UI
+    spark = SparkSession.builder.appName("data_processing").getOrCreate()
+
+    # Load data into dataframe if READ_TABLE exists
+    try:
+        df = spark.read.format("bigquery").load(READ_TABLE)
+    except Py4JJavaError as e:
+        raise Exception(f"Error reading {READ_TABLE}") from e
+
+    # Convert temperature from tenths of a degree in celsius to degrees celsius
+    df = df.withColumn("value", col("value") * 10)
```

Comment:
My brain may be broken but I'm still a little confused what this is doing. Based on your test data, this is taking f(21.0) -> 210? Maybe this makes more sense in the context of the whole demo.

Reply:
for anyone following along i've been staring at htis too long and Brad is completely right. I doubted myself in a stressed induced fervor earlier and was right about needing division all along. My elementary school math teachers would be proud.